build.sbt

```diff
@@ -1,4 +1,161 @@
-import RhoBuild._
+import sbt._
+import Keys._
+import spray.revolver.RevolverPlugin._
 
-version       in ThisBuild := s"0.17.1-SNAPSHOT"
-apiVersion    in ThisBuild <<= version.map(extractApiVersion)
+import com.typesafe.sbt.SbtGhPages.ghpages
+import com.typesafe.sbt.SbtGhPages.GhPagesKeys
+
+import com.typesafe.sbt.SbtSite.SiteKeys.siteMappings
+import com.typesafe.sbt.SbtSite.site
+
+import com.typesafe.sbt.SbtGit.git
+
+import sbtunidoc.Plugin.ScalaUnidoc
+import sbtunidoc.Plugin.unidocSettings
+import sbtunidoc.Plugin.UnidocKeys._
+
+import scala.util.Properties.envOrNone
+
+import Dependencies._, RhoPlugin._
+
+lazy val rho = project
+  .in(file("."))
+  .settings(buildSettings: _*)
+  .aggregate(`rho-core`, `rho-hal`, `rho-swagger`, `rho-examples`)
+
+lazy val `rho-core` = project
+  .in(file("core"))
+  .settings(buildSettings: _*)
+
+lazy val `rho-hal` = project
+  .in(file("hal"))
+  .settings(buildSettings :+ halDeps: _*)
+  .dependsOn(`rho-core`)
+
+lazy val `rho-swagger` = project
+  .in(file("swagger"))
+  .settings(buildSettings :+ swaggerDeps: _*)
+  .dependsOn(`rho-core` % "compile->compile;test->test")
+
+lazy val docs = project
+  .in(file("docs"))
+  .settings(buildSettings)
+  .settings(unidocSettings)
+  .settings(ghpages.settings ++ site.settings)
+  .settings(site.includeScaladoc())
+  .settings(Seq(
+    dontPublish,
+    description := "Api Documentation",
+    autoAPIMappings := true,
+    scalacOptions in Compile := scaladocOptions(
+      (baseDirectory in ThisBuild).value,
+      version.value,
+      apiVersion.value
+    ),
+    unidocProjectFilter in (ScalaUnidoc, unidoc) := inProjects(
+      `rho-core`,
+      `rho-hal`,
+      `rho-swagger`
+    ),
+    git.remoteRepo := "git@github.com:http4s/rho.git",
+    GhPagesKeys.cleanSite := VersionedGhPages.cleanSite0.value,
+    GhPagesKeys.synchLocal := VersionedGhPages.synchLocal0.value,
+    siteMappings := {
+      val (major, minor) = apiVersion.value
+      for {
+        (f, d) <- (mappings in (ScalaUnidoc, packageDoc)).value
+      } yield (f, s"api/$major.$minor/$d")
+    }
+  ))
+  .dependsOn(`rho-core`, `rho-hal`, `rho-swagger`)
+
+lazy val `rho-examples` = project
+  .in(file("examples"))
+  .settings(
+    buildSettings ++
+      Revolver.settings ++
+      Seq(
+        exampleDeps,
+        libraryDependencies ++= Seq(logbackClassic, http4sXmlInstances),
+        dontPublish
+      ): _*)
+  .dependsOn(`rho-swagger`, `rho-hal`)
+
+lazy val compileFlags = Seq("-feature") //, "-Xlog-implicits")
+
+/* Don't publish setting */
+lazy val dontPublish = packagedArtifacts := Map.empty
+
+lazy val license = licenses in ThisBuild := Seq(
+  "Apache License, Version 2.0" -> url(
+    "http://www.apache.org/licenses/LICENSE-2.0.txt")
+)
+
+lazy val buildSettings = publishing ++
+  Seq(
+    scalaVersion := "2.12.4",
+    crossScalaVersions := Seq("2.12.3", "2.11.11"),
```

Comment:
Missed a `2.12.4` here

Reply:
Do we even need `crossScalaVersions` anymore?  The only thing it buys, I think, is local running with `+` prefix.  We let Travis handle the cross builds in our other repos.

Reply:
I think its useful to have.  It allows me to test my changes fully before I push them by running '+test'.
I'll get the version updated as well.